.github/workflows/main.yml

```diff
@@ -1,8 +1,14 @@
 name: Test-cafe Health Check
 on:
+  push:
+    branches: [ 'master' ]
```

Comment:
we are running this workflow on push in master branch. I believe pushing directly on master branch should not be allowed. you should protect master branch

Reply:
Hi @Abhishek332,

Thank you for sharing your thoughts. 
I understand your perspective, but I would prefer to keep it as it is since the core maintainers use it to push changes directly into the master branch.

Let me know your thoughts.

Reply:
Are you suggesting that we should not do anything on master branch at all?

Reply:
yeah @Taz03 , we should not allow directly pushing change on master. because as code is open anybody can clone and push the changes directly on your main code.

it should be protected and only allow changes using PR

Reply:
When the PR is merged it is also considered as pushing to master, so not pushing to master would mean not merging any prs